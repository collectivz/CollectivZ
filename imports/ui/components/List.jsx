import React from 'react';
import Loader from '../components/Loader';

export default class List extends React.Component {

  constructor(props) {
    super(props);
    this.state = { isLoaded: false, isLoadable: false };
    this.renderItem = this.renderItem.bind(this);
    this.isLoaded = this.isLoaded.bind(this);
  }

  isLoaded() {
    setTimeout(() => {
      this.setState({ isLoaded: true });
    }, 850);
  }

  componentDidMount() {
    if (!this.props.isLoadable || (this.props.isLoadable && this.props.isLoadable == true)) this.setState({ isLoadable: true });
    this.isLoaded();
  }

  renderItem(item, index) {
    const {
         data,
         children,
         ...props
      } = this.props;

    props.data = item;
    props.key = index;
    const clonedChildren = children && React.cloneElement(children, props);

    return clonedChildren;
  }

  render() {
    const {
         data,
         emptyListString,
      } = this.props;

    return (
         !this.state.isLoaded && this.state.isLoadable ?
           <Loader />
            :
            (data && data.length) ?
              <div className="list">{data.map((item, index) => (this.renderItem(item, index)))}</div>
               :
               emptyListString ?
                 <div className="list-empty">
                   <p><i className="icon icon-sad" />{emptyListString}</p>
                 </div>
                  :
                 <div />
    );
  }
}
