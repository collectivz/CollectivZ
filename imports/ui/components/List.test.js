/*
import React from 'react';
import renderer from 'react-test-renderer';
import List from './List';


describe('', () => {
  const elems = [{ name: 'Group1', type: 'group' }];

  it('group snapshot', () => {
    const json = renderer.create(<List data={elems} type="group" emptyListString="Aucun group créé."><div /></List>).toJSON();
    expect(json).toMatchSnapshot();
  });
});
*/