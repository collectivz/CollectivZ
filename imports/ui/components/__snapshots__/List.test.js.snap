exports[`group snapshot 1`] = `
<div
  className="screen-box center">
  <div
    className="center-wrapper"
    id="preloader">
    <div />
  </div>
</div>
`;
